src/contexts/LikedContext.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  createContext,
+  useCallback,
+  useState,
+  useContext,
+  useEffect,
+} from "react";
+import { getData, storeData } from "../utils/asyncStorage";
+
+const LIKE_LIST_KEY = "likeList" as const;
+
+interface LikeContextType {
+  likeList: number[];
+  toggleLike: (id: number) => void;
+  isLiked: (id?: number) => boolean;
+}
+
+const LikeContext = createContext<LikeContextType>({
+  likeList: [],
+  toggleLike: () => {},
+  isLiked: () => false,
+});
+
+const LikeContextProvider = ({ children }: React.PropsWithChildren) => {
+  const [likeList, setLikedList] = useState<number[]>([]);
+  useEffect(() => {
+    (async () => {
+      const initialLikes = await getData<number[]>(LIKE_LIST_KEY);
+      if (initialLikes) {
+        setLikedList(initialLikes);
+      }
+    })();
+  }, []);
+
+  const toggleLike = useCallback((id: number) => {
+    setLikedList((prev) => {
+      if (prev.includes(id)) {
+        const nextLikeList = prev.filter((item) => item !== id);
+        storeData(LIKE_LIST_KEY, nextLikeList);
+        return nextLikeList;
+      }
+      const nextLikeList = [...prev, id];
+      storeData(LIKE_LIST_KEY, nextLikeList);
+      return nextLikeList;
```

Comment:
nit: this could be simplified to something like below, to avoid calling storeData twice (and make sure that it is called in each condition. But current code is also fine.

```suggestion
     const nextLikeList = prev.includes(id) ? prev.filter((item) => item !== id) : [...prev, id];
      storeData(LIKE_LIST_KEY, nextLikeList);
      return nextLikeList;
```